pandas/tests/io/test_pytables.py

```diff
@@ -5080,6 +5080,27 @@ def test_query_long_float_literal(self):
             expected = df.loc[[1], :]
             tm.assert_frame_equal(expected, result)
 
+    def test_query_compare_string_column(self):
+        # GH 15492
+        df = pd.DataFrame({'date': ['2014-01-01', '2014-01-02'],
+                           'real_date': date_range('2014-01-01', periods=2),
+                           'values': [1, 2]},
+                          columns=['date', 'real_date', 'values'])
+
+        with ensure_clean_store(self.path) as store:
+            store.append('test', df, format='table', data_columns=True)
+
+            ts = pd.Timestamp('2014-01-01') # noqa
+            result = store.select('test', where='real_date > ts')
+            expected = df.loc[[1], :]
+            tm.assert_frame_equal(expected, result)
+
+            for v in [2.1, True, pd.Timestamp('2014-01-01')]:
```

Comment:
can you also test int/float vs strings column (might work though.....)
Timedelta